src/main/java/uk/gov/hmcts/darts/cache/token/service/AbstractTokenCache.java

```diff
@@ -61,7 +73,7 @@ public Optional<Token> store(CacheValue value, Boolean reuseTokenIfPossible) thr
 
                 tokenToUse = lookup(value);
 
-                if (tokenToUse.isPresent() && !tokenToUse.get().valid()) {
+                if (tokenToUse.isPresent() && !tokenToUse.get().valid(true)) {
```

Comment:
struggling a bit with the logic of "! valid (true)"

Reply:
i have added an enum Token.TOKEN_EXPIRY_MODE to clear things up as @richardbull comments above